src/hotspot/share/jfr/recorder/storage/jfrStorage.cpp

```diff
@@ -291,6 +291,7 @@ void JfrStorage::register_full(BufferPtr buffer, Thread* thread) {
       JavaThread* jt = JavaThread::cast(thread);
       if (jt->thread_state() == _thread_in_native) {
         // Transition java thread to vm so it can issue a notify.
```

Comment:
I don't understand this comment. Why does issuing a notify need `WXWrite`?

Reply:
The comment is preexisting. Issuing notify doesn't need `WXWrite`. It needs a thread state transition `ThreadInVMfromNative` and that transition requires `WXWrite`.